chain/chain/src/resharding/manager.rs

```diff
@@ -191,70 +187,38 @@ impl ReshardingManager {
+        split_shard_event: &ReshardingSplitShardParams,
     ) -> Result<(), Error> {
         let block_hash = block.hash();
         let block_height = block.header().height();
+        let parent_shard_uid = split_shard_event.parent_shard;
         let _span = tracing::debug_span!(
             target: "resharding", "process_memtrie_resharding_storage_update",
             ?block_hash, block_height, ?parent_shard_uid)
         .entered();
 
-        tries.freeze_parent_memtrie(parent_shard_uid, split_shard_event.children_shards())?;
-
-        let parent_chunk_extra = self.get_chunk_extra(block_hash, &parent_shard_uid)?;
-        let boundary_account = split_shard_event.boundary_account;
-
-        let mut trie_store_update = self.store.store_update();
+        let parent_chunk_extra =
+            self.store.chain_store().get_chunk_extra(block_hash, &parent_shard_uid)?;
+        let mut store_update = self.store.trie_store().store_update();
 
         // TODO(resharding): leave only tracked shards.
         for (new_shard_uid, retain_mode) in [
             (split_shard_event.left_child_shard, RetainMode::Left),
             (split_shard_event.right_child_shard, RetainMode::Right),
         ] {
-            let Some(memtries) = tries.get_memtries(new_shard_uid) else {
-                tracing::error!(
-                    "Memtrie not loaded. Cannot process memtrie resharding storage
-                     update for block {:?}, shard {:?}",
-                    block_hash,
-                    parent_shard_uid,
-                );
-                return Err(Error::Other("Memtrie not loaded".to_string()));
-            };
-
-            tracing::info!(
-                target: "resharding", ?new_shard_uid, ?retain_mode,
-                "Creating child memtrie by retaining nodes in parent memtrie..."
-            );
-            let mut memtries = memtries.write().unwrap();
-            let mut trie_recorder = TrieRecorder::new(None);
-            let mode = TrackingMode::RefcountsAndAccesses(&mut trie_recorder);
-            let memtrie_update = memtries.update(*parent_chunk_extra.state_root(), mode)?;
-
-            let trie_changes = memtrie_update.retain_split_shard(
-                &boundary_account,
-                retain_mode,
-                AccessOptions::DEFAULT,
-            );
-            Self::duplicate_nodes_at_split_boundary(
-                &mut trie_store_update.trie_store_update(),
-                trie_recorder.recorded_iter(),
-                parent_shard_uid,
-            );
-            let memtrie_changes = trie_changes.memtrie_changes.as_ref().unwrap();
-            let new_state_root = memtries.apply_memtrie_changes(block_height, memtrie_changes);
-            drop(memtries);
+            tracing::info!(target: "resharding", ?new_shard_uid, ?retain_mode, "Splitting parent memtrie");
+            let parent_trie = tries
+                .get_trie_for_shard(parent_shard_uid, *parent_chunk_extra.state_root())
```

Comment:
What happened to error handling? Did you change the semantics from "if no memtrie then error" to "if no memtrie use disk trie"?